src/main/java/co/alphavantage/output/exchange/CurrencyExchange.java

```diff
@@ -0,0 +1,77 @@
+package co.alphavantage.output.exchange;
+
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import co.alphavantage.output.AlphaVantageException;
+import co.alphavantage.output.JsonParser;
+import co.alphavantage.output.exchange.data.CurrencyExchangeData;
+
+public class CurrencyExchange {
+	private CurrencyExchangeData data;
+	
+	CurrencyExchange(CurrencyExchangeData data){
+		this.data = data;
+	}
+	
+
+	public CurrencyExchangeData getData() {
+		return data;
+	}
+
+
+  /**
+   * Create Currency Exchange data representation from json object
+   * @param json string to parse
+   * @return Currency Exchange data
+   * @throws AlphaVantageException
+   */
+	public static CurrencyExchange from(String json) throws AlphaVantageException {
+		CurrencyExchangeParser parser = new CurrencyExchangeParser();
+		return parser.parseJson(json);
+	}
+
+  /**
+   * Helper class for parsing json to {@code Currency Exchange}
+   */
+	private static class CurrencyExchangeParser extends JsonParser<CurrencyExchange>{
+		public CurrencyExchange resolve(JsonObject rootObject) {
+			  Type dataType = new TypeToken<Map<String, Map<String, String>>>() {
+		      }.getType();
+		      try {
+		        Map<String, Map<String, String>> data = GSON.fromJson(rootObject, dataType);
+		        List<CurrencyExchangeData> exchanges = new ArrayList<>();
+		        data.forEach((key, values) -> {
+		            exchanges.add(createCurrencyExchangeData(values));
+		        });
+			    return new CurrencyExchange(exchanges.get(0));
+		      } catch (JsonSyntaxException e) {
+		        throw new AlphaVantageException("technical indicators api change", e);
+		      }
+		}
+		
+		private CurrencyExchangeData createCurrencyExchangeData(Map<String, String> values) throws JsonSyntaxException {
+	        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

```

Comment:
You dont need to create a new instance of DateTimeFormatter, there is one to reuse in JsonParser :-)

Reply:
I will review it. ;)

Reply:
@braisly good one. I can see java8's way is recommended even by the joda time guys. Updating seems like the natural path.

Reply:
@Braisly Yes we should change that. Good catch!